feat: add filesystem hierarchical order to stack orchestration (#484)
mineiros-io/terramate: docs/orchestration.md

* 1 - **stack A**
* 2 - **stack B**

To help with that terramate provides two ways to define the desired order of
execution between stacks.

### Filesystem hierarchical order

> By creating an hierarchy of stacks in the filesystem will make Terramate execute
> the parent stacks before the children ones.

Reviewer: ```suggestion
Creating an hierarchy of stacks in the filesystem will make Terramate execute
parent stacks before their children.
```

This is a double-edged sword because it allows you to move stacks around to
better resemble the cloud infrastructure and the order of which things must
be executed with zero code change but in counterpart beware that moving stacks 

> just for cosmetic purposes can potentially change order and break stuff.

Reviewer: ```suggestion
just for cosmetic purposes can potentially change the execution order and break things.
```

### Explicit Order Of Execution

Order of execution can be explicitly declared inside the **stack** block using

> the field **before** and **after**. 

Reviewer: ```suggestion
the fields **before** and **after**. 
```

Each field is a set of strings (**set(string)**), where each string is a 
reference to another stack.

The explicit order can be used in conjunction with the implicit filesystem order
but have in mind that a parent stack in the filesystem can never execute after a
child one, and trying to make this using explicit **before** and **after** clauses

> will lead to a cycle error.

Reviewer: nice
